packages/good-design/src/advanced/web3action/Web3Action.tsx

```diff
@@ -81,7 +81,7 @@ export const Web3ActionButton = ({
   const switchToChain = useCallback(
     async (chain: number) => {
       const switchFn = switchChain || switchNetwork;
-      const result = switchFn(chain).catch(throwCancelled);
+      const result = await switchFn(chain).catch(throwCancelled);
```

Comment:
Why changes in action button again ? We already fixed them all
Please revert

Reply:
because I tested again and the loading text did not get updated anymore